examples/src/main/java/kafka/examples/KafkaConsumerProducerDemo.java

```diff
@@ -32,8 +32,14 @@ public static void main(String[] args) throws InterruptedException {
         Consumer consumerThread = new Consumer(KafkaProperties.TOPIC, "DemoConsumer", Optional.empty(), false, 10000, latch);
         consumerThread.start();
 
-        if (!latch.await(5, TimeUnit.MINUTES)) {
-            throw new TimeoutException("Timeout after 5 minutes waiting for demo producer and consumer to finish");
+        try {
+            if (!latch.await(5, TimeUnit.MINUTES)) {
+                throw new TimeoutException("Timeout after 5 minutes waiting for demo producer and consumer to finish");
+            }
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } finally {
+            consumerThread.shutdown();
```

Comment:
Could we combine the finally block with L45-46? Also I was thinking whether we should close the producer thread as well.